viewing_party/main.py

```diff
@@ -0,0 +1,179 @@
+    friends_movies = []
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            friends_movies.append(movie)
+
+    for movie in user_data["watched"]:
+        if movie not in friends_movies:
+            users_unique_movies.append(movie)
+    
+    return users_unique_movies
+
+def get_friends_unique_watched(user_data):
+
+    friends_unique_movies = []
+    users_movies = []
+
+    for movie in user_data["watched"]:
+        users_movies.append(movie)
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie not in users_movies:
+                if movie not in friends_unique_movies:
+                    friends_unique_movies.append(movie)
+    
+    return friends_unique_movies
+
+
+# ---------------------Wave 4---------------------------
+
+def get_available_recs(user_data):    
+
+    friends_movies = user_data["friends"]
+    subscriptions = user_data["subscriptions"]
+    recommendations = []
+
+    for friend in friends_movies:
+        for movie in friend["watched"]:
+            if movie not in user_data["watched"] and\
+            movie not in recommendations and\
+            movie["host"] in subscriptions:
+                recommendations.append(movie)
+
+    for movie in recommendations:
+        # if host, service not in recommendations:
+        if movie["host"] not in subscriptions:
+            recommendations.remove(movie)
+    # if len(user_data["watched"]) == 0:
+    #     return []
+
+    return recommendations
+
+# ---------------------Wave 5---------------------------
+
+def get_new_rec_by_genre(user_data):
+
+    # assign results of get_most_watched_genre to a variable
+    user_genre = get_most_watched_genre(user_data)
+    friends_movies = user_data["friends"]
+    recommendations = []
+
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie["genre"] == user_genre:
+                recommendations.append(movie)
+    
+    return recommendations
+
+def get_rec_from_favorites(user_data):
+
+    user_favories = user_data["favorites"]
+    friends_movies = user_data["friends"]
+    favorites = []
+    user_rec = []
+
+    for movie in user_favories:
+        favorites.append(movie)
+
+    for movie in favorites:
+        for movie in friends_movies["watched"]:
+            # if movie not in friends_movies["watched"]:
+            user_rec.append(movie)
+
+    return user_rec
+
+def get_rec_from_favorites(user_data):
+
+    user_favories = user_data["favorites"]
+    friends_movies = user_data["friends"]
+    favorites = []
+    friend_watched = []
+    user_rec = []
+
+    for movie in user_favories:
+        favorites.append(movie)
+
+    for friend in friends_movies:
+        for movie in friend["watched"]:
+            friend_watched.append(movie)
+
+    for movies in favorites:
+        if movies not in friend_watched:
+            user_rec.append(movies)
+    
+    return user_rec
```

Comment:
I'm curious why this same function is defined twice. 
In this case the first function definition will just get overwritten by the 2nd one.